image-builder/image_builder.sh

```diff
@@ -113,55 +125,96 @@ OK "init is installed"
 OK "Agent installed"
 [ "$(id -u)" -eq 0 ] || die "$0: must be run as root"
 
+ROOTFS_SIZE=$(du -B 1MB -s "${ROOTFS}" | awk '{print $1}')
 BLOCK_SIZE=${BLOCK_SIZE:-4096}
-IMG_SIZE=${IMG_SIZE:-80}
-
-info "Creating raw disk with size ${IMG_SIZE}M"
-qemu-img create -q -f raw "${IMAGE}" "${IMG_SIZE}M"
-OK "Image file created"
-
-# Kata runtime expect an image with just one partition
-# The partition is the rootfs content
+OLD_IMG_SIZE=0
 
-info "Creating partitions"
-parted "${IMAGE}" --script "mklabel gpt" \
-"mkpart ext4 1M -1M"
-OK "Partitions created"
+# Extra buffer space for the raw disk in percentage.
+IMG_BUFFER_PERCENT=30
 
-# Get the loop device bound to the image file (requires /dev mounted in the
-# image build system and root privileges)
-DEVICE=$(losetup -P -f --show "${IMAGE}")
-
-#Refresh partition table
-partprobe "${DEVICE}"
+# Calculate image size based on the rootfs
+calculate_img_size()
+{
+    IMG_SIZE=${IMG_SIZE:-$(((($ROOTFS_SIZE*$IMG_BUFFER_PERCENT)/100)+$ROOTFS_SIZE))}
+}
 
-MOUNT_DIR=$(mktemp -d osbuilder-mount-dir.XXXX)
-info "Formating Image using ext4 format"
-mkfs.ext4 -q -F -b "${BLOCK_SIZE}" "${DEVICE}p1"
-OK "Image formated"
+# Cleanup
+cleanup()
+{
+    sync
+    umount -l ${MOUNT_DIR}
+    rm -rf ${MOUNT_DIR}
```

Comment:
I think this can be `rm -f ${MOUNT_DIR}`?

Reply:
It's a directory.

Reply:
Sorry, yes. I meant to say `rmdir ${MOUNT_DIR}`. The directory is empty so this is potentially safer.